tools/mason/MasonSearch.chpl

```diff
@@ -66,16 +69,37 @@ proc masonSearch(origArgs : [] string) {
         }  else {
           const ver = findLatest(searchDir + dir);
           const versionZero = new VersionInfo(0, 0, 0);
-
           if ver != versionZero then
             results.push_back(name + " (" + ver.str() + ")");
+            showResults.push_back(name);
+            versionToReturn.push_back(ver.str());
         }
       }
     }
   }
 
   for r in results.sorted() do writeln(r);
 
+  if results.size == 1 && show {
+    writeln('Displaying the latest version: ' + showResults + '@' + versionToReturn);
+    showToml(versionToReturn + '.toml', showResults);
+  }
+
+  if results.size == 0 && show {
+  writeln('"' + query + '" returned no packages');
+  writeln('--show requires the search to return one package');
+  }
+
+  if show && results.size > 1 {
+    if query == '.*' {
```

Comment:
What special case is this handling, when `mason search` is given no args?

Reply:
When `mason search --show` is given no args the script subs in `.*` as the query to return all of the packages, without this case it would return `.* returned more than one package`. My thinking was that would be a little confusing and opted to create a case for a general error when there is no query with --show.